x-pack/plugins/apm/public/components/app/mobile/service_overview/filters/index.tsx

```diff
@@ -9,45 +9,52 @@ import {
   EuiFlexGroupProps,
   EuiFlexItem,
   EuiSelect,
+  EuiLoadingSpinner,
 } from '@elastic/eui';
 import React from 'react';
 import { useHistory } from 'react-router-dom';
-import { Environment } from '../../../../../../common/environment_rt';
 import { useApmServiceContext } from '../../../../../context/apm_service/use_apm_service_context';
+import { useAnyOfApmParams } from '../../../../../hooks/use_apm_params';
 import { useBreakpoints } from '../../../../../hooks/use_breakpoints';
-import { useFetcher } from '../../../../../hooks/use_fetcher';
+import { useFetcher, isPending } from '../../../../../hooks/use_fetcher';
+import { useTimeRange } from '../../../../../hooks/use_time_range';
 import type { APIReturnType } from '../../../../../services/rest/create_call_apm_api';
 import { push } from '../../../../shared/links/url_helpers';
 
 type MobileFilter =
   APIReturnType<'GET /internal/apm/services/{serviceName}/mobile/filters'>['mobileFilters'][0];
 
-interface Props {
-  end: string;
-  environment: Environment;
-  transactionType?: string;
-  kuery: string;
-  start: string;
-  filters: Record<MobileFilter['key'], string | undefined>;
-}
-
 const ALL_OPTION = {
   value: 'all',
   text: 'All',
 };
 
-export function MobileFilters({
-  end,
-  environment,
-  transactionType,
-  kuery,
-  start,
-  filters,
-}: Props) {
+export function MobileFilters() {
   const history = useHistory();
   const { isSmall, isLarge } = useBreakpoints();
   const { serviceName } = useApmServiceContext();
-  const { data = { mobileFilters: [] } } = useFetcher(
+
+  const {
+    query: {
+      environment,
+      kuery,
+      rangeFrom,
+      rangeTo,
+      netConnectionType,
+      device,
+      osVersion,
+      appVersion,
+      transactionType,
+    },
+  } = useAnyOfApmParams(
```

Comment:
`useApmParams` supports the possibility of passing `*` but in this case, I'm passing only the pages that send the mobile filter params. Otherwise, I'll get a type error. 

For example, the Alerts page (`/mobile-services/{serviceName}/alerts`) won't support mobile filters

Reply:
Thanks for clarifying